### PR TITLE
Improvements to `EitherOrBoth`

### PR DESCRIPTION
**Pull request by JoJoJet**
* Added more methods for accessing the left and right variants.
* Added methods for inserting left and right values (using unsafe code).
* Improved some grammar.

### CODE REVIEW

**Comment by phimuemue** [comment]
Hi there, thanks for addressing the compilation issues.

Could you rebase the changes so we do not need to review back-and-forth changes?

**Comment by JoJoJet** [comment]
> Could you rebase the changes so we do not need to review back-and-forth changes?

Done, I think.

File: src/either_or_both.rs [hunk 1]
```diff
@@ -42,36 +44,107 @@ impl<A, B> EitherOrBoth<A, B> {
...
+    /// If `Both`, return `Some` containing the left and right values. Otherwise, return `None`.
     pub fn both(self) -> Option<(A, B)> {
         match self {
             Both(a, b) => Some((a, b)),
             _ => None,
         }
     }
 
+    /// If `Left` or `Both`, return the left value. Otherwise, convert the right value and return it.
+    pub fn into_left(self) -> A
```
Comment (oxalica) [inline src/either_or_both.rs:127]: I prefer `left_biased`, which implies the fallback behavior on `Right`. So does `into_right` -> `right_biased`.
This `*_biased` wording is also used by `rowan`, the parser library of `rust-analyzer`, in an enum like our `EitherOrBoth` but allows empty.
https://docs.rs/rowan/0.15.11/rowan/enum.TokenAtOffset.html#method.left_biased

**Review by oxalica — CHANGES_REQUESTED** [review]

File: src/either_or_both.rs (hunk 1 above)
Comment (oxalica) [reply src/either_or_both.rs:127]: > I don't think that the `*_biased` naming convention really captures what's going on here.

To my intuition, `left_biased` implies that "prefer Left, but fallback to return Right if there's no Left". `into_left` sounds like "we know there's a Left inside, just unwrap it" (like `into_inner`?).
But you're right, there's a conversion when `Right(_)` is encountered, and `into_left` captures this behavior.

**Review by jswrenn — APPROVED** [review]
bors r+

**Pull request merged**